client/codegen_pinvoke/Main.cs

```diff
@@ -7,55 +7,77 @@
 using DirectoryInfo = System.IO.DirectoryInfo;
 using Path = System.IO.Path;
 using Dir = System.IO.Directory;
+using System.Collections.Generic;
 
 
 namespace Codegen
 {
 
+    public record LibInfo
+    {
+        public DirectoryInfo output_dir { get; init; }
+        public DirectoryInfo input_dir { get; init; }
+        public string crate_name { get; init; }
+
+        public LibInfo(string output_dir, string input_dir, string crate_name)
+        {
+            this.output_dir = new DirectoryInfo(output_dir);
+            this.input_dir = new DirectoryInfo(input_dir);
+            this.crate_name = crate_name;
+        }
+    }
+
     public class Codegen : CppSharp.ILibrary
     {
         static DirectoryInfo project_dir = GetProjectDir();
-        DirectoryInfo output_dir;
+        LibInfo lib_info;

```

Comment:
Mark `private`

Reply:
sure - keep in mind though that this is an application not a library

Reply:
👍 just best practice in C# to mark private & use getter/setter if needed outside of the class.

Reply:
done